addons/account/tests/test_account_account.py

```diff
@@ -342,3 +342,45 @@ def test_account_name_onchange(self):
         # saving a form without a code should not be possible
         with self.assertRaises(AssertionError):
             account_form.save()
+
+
+    def test_generate_account_suggestions(self):
+
+        # Setting up
+        partner = self.env['res.partner'].create({
+            'name': 'partner_test_generate_account_suggestions',
+        })
+        account = self.company_data['default_account_revenue']
+        product = self.env['product.product'].create({
+            'name': 'Product_test_generate_account_suggestions',
+            'default_code': 'Ptgas',
+            'lst_price': 100.0,
+            'standard_price': 100.0,
+        })
+        self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': partner.id,
+            'invoice_date': '9998-12-31',  # Test will brake after year 9999
+            'invoice_line_ids': [Command.create({
+                'product_id': product.id,
+                'price_unit': 1000,
+            })]
+        })
```

Comment:
It's not necessary to do the whole setup, you could use the existing self.partner_a and just deprecate some of its most frequent accounts